### ISSUE TITLE
Request too large!!

### ISSUE THREAD

**Issue by vecna** [tracking-exposed/facebook#58]
![image](https://cloud.githubusercontent.com/assets/89555/23709565/a6b15aa4-03f8-11e7-908b-3ce3c70ce8e4.png)

loading many posts at one reach the server limit of 3Mb

**Comment by konarkmodi** [comment]
I would suggest, that instead of sending data in batches, we split them. To keep it simple, send each impression as a different request.

**Comment by vecna** [comment]
Yes split them looks like the simplest way to avoid this issue. but send each impression as a different request would mean change the internal structure of the extension, it is too much.

We have setup an internal timeout in the extension, https://github.com/tracking-exposed/web-extension/blob/master/webpack.config.js#L44 (in this line, when is development is 10seconds, default is 20 seconds delay)

The timing can be reduced to 5 seconds in production perhaps and this would cause, in 5 seconds, facebook do not provide you more than 6-8 post top..

**Comment by vecna** [comment]
as context: each post might contain up to 30kbytes, and before the limit was set to 1mb. In certain condition (large monitor, large performance) the threshold of 1mb get reach and so I extend it to 3mb

**Comment by vrde** [comment]
We can add a size boundary in the client, so it flushes if a timeout is reached **or** a specific size is reached.

**Comment by vecna** [comment]
![image](https://user-images.githubusercontent.com/89555/40125096-56857a64-592a-11e8-9863-49fef29a1c6e.png)
